src/query/expression/src/types/string.rs

```diff
@@ -284,10 +284,24 @@ impl StringColumnBuilder {
     }
 
     pub fn append(&mut self, other: &Self) {
-        self.data.extend_from_slice(&other.data);
+        // the first offset of other column may not be zero
```

Comment:
```suggestion
    pub fn append(&mut self, other: &Self) {
        // the first offset of other column may not be zero
        let start = self.offsets.last().cloned().unwrap() - *other.offsets.first().unwrap();
        self.data.extend_from_slice(&other.data);
        self.offsets
            .extend(other.offsets.iter().skip(1).map(|offset| start + offset));
    }
```

Reply:
`other.data` needs to slice by other start.

Reply:
`start - other_start` may overflow